autogen.sh: temporarily disable shm counters in mongo-c-driver

bkil-syslogng opened the pull request:
This is a temporary quick workaround to help testing, do not merge yet.

* TODO: extract mongo-c-driver kludges to a separate function
* TODO: introduce configure option to reenable shm counters with warnings

czanik commented:
Tested (patched latest git head, created dist file, recompiled rpm package and tested the result), and it solved the problem.

bazsi commented:
I think this could go in as it is. It fixes a reported security issue... I have filed this bug in mongo-c-driver, let's see how that timeline works out and either update our in-tree copy, or just merge the workaround.

bazsi commented:
https://jira.mongodb.org/plugins/servlet/mobile#issue/CDRIVER-1691/comment/1405406

They said they could fix this in their upcoming version.

bkil-syslogng commented:
@bazsi
Thanks for not letting this down. I rebased to master and adjusted the commit message (please review whether it is okay), so it should still pass Travis CI. If it does not, please press the retry button...

I agree that it could go in if it does no great harm (other than breaking mongo-stats). We can commit further improvements later on, hopefully before the next release.

bazsi commented:
@czanikp pls notify the original reporter.
